File: IQ/Classes/clsServiceLevel.vb
Option Explicit On
Option Strict On

Imports dataAccess
Imports System.IO

' Represents an HP Service Pack service level

Public Class clsServiceLevel
    Implements i_Editable

    Public ID As Integer
    Public MfrCode As String
    Public ServiceLevel As Integer
    Public ServiceLevelGroup As String
    Public SuperGroup As String
    Public Description As clsTranslation
    Public Duration As Integer
    Public PostWarranty As Boolean
    Public Disabled As Boolean
    Public ServiceType As clsServiceType
    Public Response As clsResponse
    Public HpeDmr As Boolean
    Public HpeCdmr As Boolean
    Public HpiAdp As Boolean
    Public HpiDmr As Boolean
    Public HpiTravel As Boolean
    Public HpiTracing As Boolean
    Public HpiTheft As Boolean

    Private Const TABLE As String = "ServiceLevelMap"

    Public Sub New()

    End Sub

    Public Sub New(id As Integer, mfrCode As String, serviceLevel As Integer, serviceLevelGroup As String, superGroup As String, description As clsTranslation, duration As Integer, postWarranty As Boolean,
                   disabled As Boolean, serviceType As clsServiceType, response As clsResponse, hpeDmr As Boolean, hpeCdmr As Boolean, hpiAdp As Boolean, hpiDmr As Boolean, hpiTravel As Boolean, hpiTracing As Boolean, hpiTheft As Boolean)

        Me.ID = id
        Me.MfrCode = mfrCode
        Me.ServiceLevel = serviceLevel
        Me.ServiceLevelGroup = serviceLevelGroup
        Me.SuperGroup = superGroup
        Me.Description = description
        Me.Duration = duration
        Me.PostWarranty = postWarranty
        Me.Disabled = disabled
        Me.ServiceType = serviceType
        Me.Response = response
        Me.HpeDmr = hpeDmr
        Me.HpeCdmr = hpeCdmr
        Me.HpiAdp = hpiAdp
        Me.HpiDmr = hpiDmr
        Me.HpiTravel = hpiTravel
        Me.HpiTracing = hpiTracing
        Me.HpiTheft = hpiTheft

    End Sub

    Public Function Insert(ByRef errormessages As List(Of String)) As Object Implements i_Editable.Insert

        Return New clsServiceLevel(Me.ID, Me.MfrCode, Me.ServiceLevel, Me.ServiceLevelGroup, Me.SuperGroup, Me.Description, Me.Duration, Me.PostWarranty, Me.Disabled, Me.ServiceType, Me.Response, Me.HpeDmr, Me.HpeCdmr, Me.HpiAdp, Me.HpiDmr, Me.HpiTravel, Me.HpiTracing, Me.HpiTheft)

    End Function

    Public Sub delete(ByRef errorMessages As List(Of String)) Implements i_Editable.delete

        Dim sql$ = String.Format("delete from {0} where id={1}", TABLE, Me.ID)

        da.DBExecutesql(sql$)

    End Sub

    Public Sub Update(ByRef errorMessages As List(Of String)) Implements i_Editable.update

        Dim sql$ = String.Format("update {0} set mfrCode='{1}', ServiceLevel={2}, ServiceLevelGroup='{3}', SuperGroup='{4}', Duration={5}, PostWarranty={6}, Disabled={7}, ServiceType={8}, Response={9}, DMR={10}, HpeDmr={11}, HpeCdmr={12}, HpiAdp={13}, HpiDmr={14}, HpiTravel={15}, HpiTracining={16}, HpiTheft={17) where ID={18}",
                                 TABLE, Me.MfrCode, Me.ServiceLevel, Me.ServiceLevelGroup, Me.SuperGroup, Me.Duration, Me.PostWarranty, Me.Disabled, Me.ServiceType.ID, Me.Response.ID, Me.HpiAdp, Me.HpeDmr, Me.HpeCdmr, Me.HpiDmr, Me.HpiTravel, Me.HpiTracing, Me.HpiTheft, Me.ID)

        da.DBExecutesql(sql$, False)

    End Sub

    Public Function DisplayName(language As clsLanguage) As String Implements i_Editable.displayName

        Return Me.Description.text(language)

    End Function

    Public ReadOnly Property Manufacturer As Manufacturer

        Get

            Manufacturer = Manufacturer.Unknown

            If Not String.IsNullOrEmpty(Me.mfrCode) Then
                If String.Equals(Me.mfrCode, "HPI", StringComparison.InvariantCultureIgnoreCase) Then
                    Manufacturer = Manufacturer.HPI
                ElseIf String.Equals(Me.mfrCode, "HPE", StringComparison.InvariantCultureIgnoreCase) Then
                    Manufacturer = Manufacturer.HPE
                End If
            End If

        End Get

    End Property

End Class

File: IQ/Classes/clsServiceLevel.cs
using dataAccess;
using System.IO;



// Represents an HP Service Pack service level

public class clsServiceLevel : i_Editable
{

    public int ID;
    public string MfrCode;
    public int ServiceLevel;
    public string ServiceLevelGroup;
    public string SuperGroup;
    public clsTranslation Description;
    public int Duration;
    public bool PostWarranty;
    public bool Disabled;
    public clsServiceType ServiceType;
    public clsResponse Response;
    public bool HpeDmr;
    public bool HpeCdmr;
    public bool HpiAdp;
    public bool HpiDmr;
    public bool HpiTravel;
    public bool HpiTracing;
    public bool HpiTheft;

    private const string TABLE = "ServiceLevelMap";

    public clsServiceLevel()
    {

    }

    public clsServiceLevel(int id, string mfrCode, int serviceLevel, string serviceLevelGroup, string superGroup, clsTranslation description, int duration, bool postWarranty, bool disabled, clsServiceType serviceType, clsResponse response, bool hpeDmr, bool hpeCdmr, bool hpiAdp, bool hpiDmr, bool hpiTravel, bool hpiTracing, bool hpiTheft)
    {

        this.ID = id;
        this.MfrCode = mfrCode;
        this.ServiceLevel = serviceLevel;
        this.ServiceLevelGroup = serviceLevelGroup;
        this.SuperGroup = superGroup;
        this.Description = description;
        this.Duration = duration;
        this.PostWarranty = postWarranty;
        this.Disabled = disabled;
        this.ServiceType = serviceType;
        this.Response = response;
        this.HpeDmr = hpeDmr;
        this.HpeCdmr = hpeCdmr;
        this.HpiAdp = hpiAdp;
        this.HpiDmr = hpiDmr;
        this.HpiTravel = hpiTravel;
        this.HpiTracing = hpiTracing;
        this.HpiTheft = hpiTheft;

    }

    public dynamic Insert(ref List<string> errormessages)
    {

        return new clsServiceLevel(this.ID, this.MfrCode, this.ServiceLevel, this.ServiceLevelGroup, this.SuperGroup, this.Description, this.Duration, this.PostWarranty, this.Disabled, this.ServiceType, this.Response, this.HpeDmr, this.HpeCdmr, this.HpiAdp, this.HpiDmr, this.HpiTravel, this.HpiTracing, this.HpiTheft);

    }

    public void delete(ref List<string> errorMessages)
    {

        System.Char sql = string.Format("delete from {0} where id={1}", TABLE, this.ID);

        da.DBExecutesql(sql);

    }

    public void update(ref List<string> errorMessages)
    {

        System.Char sql = string.Format("update {0} set mfrCode=\'{1}\', ServiceLevel={2}, ServiceLevelGroup=\'{3}\', SuperGroup=\'{4}\', Duration={5}, PostWarranty={6}, Disabled={7}, ServiceType={8}, Response={9}, DMR={10}, HpeDmr={11}, HpeCdmr={12}, HpiAdp={13}, HpiDmr={14}, HpiTravel={15}, HpiTracining={16}, HpiTheft={17) where ID={18}",
            TABLE, this.MfrCode, this.ServiceLevel, this.ServiceLevelGroup, this.SuperGroup, this.Duration, this.PostWarranty, this.Disabled, this.ServiceType.ID, this.Response.ID, this.HpiAdp, this.HpeDmr, this.HpeCdmr, this.HpiDmr, this.HpiTravel, this.HpiTracing, this.HpiTheft, this.ID);

        da.DBExecutesql(sql, false);

    }

    public string displayName(clsLanguage language)
    {

        return this.Description.text(language);

    }

    public Manufacturer Manufacturer
    {

        get
        {
            Manufacturer returnValue = default(Manufacturer);

            returnValue = returnValue.Unknown;

            if (!string.IsNullOrEmpty(this.MfrCode))
            {
                if (string.Equals(this.MfrCode, "HPI", StringComparison.InvariantCultureIgnoreCase))
                {
                    returnValue = returnValue.HPI;
                }
                else if (string.Equals(this.MfrCode, "HPE", StringComparison.InvariantCultureIgnoreCase))
                {
                    returnValue = returnValue.HPE;
                }
            }

            return returnValue;
        }

    }

}